File: KotlinAndroid/app/src/main/kotlin/kotlin/sample/MainActivity.kt
package kotlin.sample

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.Menu
import android.view.MenuItem


//  Android Extension
import kotlinx.android.synthetic.activity_main.*

/**
 * Created by SidneyXu on 2015/11/25.
 */
public class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        button1.setOnClickListener {
            navigate<CountryListActivity>()
        }
    }

    //  Reified types can get class not class type
    inline public fun <reified T : Activity> Activity.navigate(bundle: Bundle? = null) {
        val intent = Intent(this, T::class.java)
        if (bundle != null) {
            intent.putExtras(bundle)
        }
        startActivity(intent)
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        // Inflate the menu; this adds items to the action bar if it is present.
        menuInflater.inflate(R.menu.menu_main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem?): Boolean {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        val id = item!!.itemId

        //noinspection SimplifiableIfStatement
        if (id == R.id.action_settings) {
            return true
        }

        return super.onOptionsItemSelected(item)
    }


}

File: ScalaAndroid/app/src/main/scala/scala/sample/MainActivity.scala
package scala.sample

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.{Menu, MenuItem, View}
import android.widget.Button
import org.scaloid.common._

class MainActivity extends AppCompatActivity with SActivity {

    protected override def onCreate(savedInstanceState: Bundle) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        find[Button](android.R.id.button1).onClick((v: View) =>
            startActivity[CountryListActivity]
        )
    }

    override def onCreateOptionsMenu(menu: Menu): Boolean = {
        getMenuInflater.inflate(R.menu.menu_main, menu)
        true
    }

    override def onOptionsItemSelected(item: MenuItem): Boolean = {
        val id: Int = item.getItemId
        if (id == R.id.action_settings) {
            return true
        }
        super.onOptionsItemSelected(item)
    }
}
